webknossos/webknossos/dataset/_array.py

```diff
@@ -26,12 +25,25 @@ def _is_power_of_two(num: int) -> bool:
 def _fsstore_from_path(path: Path, mode: str = "a") -> FSStore:
     storage_options = {}
     if hasattr(path, "_kwargs"):
+        path = cast(UPath, path)
         storage_options = path._kwargs.copy()
         del storage_options["_url"]
 
     return FSStore(url=str(path), mode=mode, **storage_options)
 
 
+@contextmanager
+def _blosc_fix_threading() -> Iterator[None]:
```

Comment:
From https://zarr.readthedocs.io/en/stable/tutorial.html#parallel-computing-and-synchronization
> When using multiple processes to parallelize reads or writes on arrays using the Blosc compression library, it may be necessary to set numcodecs.blosc.use_threads = False, as otherwise Blosc may share incorrect global state amongst processes causing programs to hang. See also the section on [Configuring Blosc](https://zarr.readthedocs.io/en/stable/tutorial.html#tutorial-tips-blosc) below.

I think would be wise to pick the safe option here and deactivate blosc-internal parallelism. I haven't encountered the issue during testing, tho.